GAE/src/com/gallatinsystems/survey/dao/SurveyUtils.java

```diff
@@ -101,12 +106,44 @@ public static Survey copySurvey(Survey source, SurveyDto dto) {
         return newSurvey;
     }
 
+    
+    /**
+     * @param sourceGroup
+     * @param copyGroup
+     * @param newSurveyId
+     * @param qDependencyResolutionMap
+     * @return
+     * 
+     * copies a question group to another survey or within the same survey (which risks creating duplicated question ids).
+     */
     public static QuestionGroup copyQuestionGroup(QuestionGroup sourceGroup,
             QuestionGroup copyGroup, Long newSurveyId, Map<Long, Long> qDependencyResolutionMap) {
 
         final QuestionDao qDao = new QuestionDao();
         final Long sourceGroupId = sourceGroup.getKey().getId();
         final Long copyGroupId = copyGroup.getKey().getId();
+        final boolean sameSurvey = sourceGroup.getSurveyId() == newSurveyId;
+        Set<String> idsInUse = null;
+        
+        if (sameSurvey) {
```

Comment:
Having this block within the `copyGroup` function,means we'll make redundant queries for these entities when copying an entire survey and will definitely have a performance impact.  Can we instead update the method signature of the copy group to take in a `Set<String> questionIdsList` parameter that we provide from outside of the function?

It will also serve the purpose of us checking whether we need to even verify the question ids. i.e. in some cases we are copying surveys without question ids but if we leave it like this, this code will still be executed for those cases.